PayButton/paybutton-server
# [Issue #334] feat: Updating paybuttons txs automatically

Depends on
---
- [x] #505 

Description
---
Solves #334 for paybuttons. Live update transactions on the paybutton detail view.


Test plan
---
1. Rerun the containers
2. Go to a paybutton with one of your addresses (create one if needed)
3. Send some XEC to it. You should very quickly see the unconfirmed transaction there, without the needing to refresh the page.
4. When the transaction is confirmed, this also should be updated immediately.


Technical Remarks
---
- I changed the behavior introduced in #505 so that we don't fetch all txs again when a new one arrives, but simply use the data from the new one to update all clients. This means the updating the client requires no DB access.
- I refactored the `PaybuttonDetail` component to be a functional component instead of a class component. The reasoning is that not only this is encouraged in the current [react docs](https://react.dev/reference/react/Component), but also was easier to setup the cleanup for the SSE clients when using `useEffect`.
